micronota/parsers/gff.py

```diff
@@ -0,0 +1,297 @@
+    if _too_many_blanks(fh, 5):
+        return False, {}
+
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    if line.startswith('##gff-version 3'):
+        return True, {}
+    else:
+        return False, {}
+
+
+def _is_float(input):
+    try:
+        float(input)
+    except ValueError:
+        return False
+    return True
+
+
+def _parse_attr(s):
+    _field = s.split(';')
+    _attr = ()
+    for f in _field:
+        try:
+            _type, _val = f.split('=')
+        except ValueError:
+            _type, _val = 0, 0
+
+        _attr = _attr + (_type, _val)
+    return _attr
+
+
+def _parse_required(s):
+    if s.isdigit():
+        return int(s)
+    elif _is_float(s):
+        return float(s)
+    else:
+        return s
+
+
+def _construct(record, constructor=None, **kwargs):
+    if constructor is None:
+        constructor = IntervalMetadata
+    if constructor is IntervalMetadata:
+        return IntervalMetadata(features=record)
+
+
+@gff.reader(None)
+def _gff_to_generator(fh, constructor=None, **kwargs):
+    for record in _parse_records(fh):
+        yield _construct(record, constructor, **kwargs)
+
+
+@gff.reader(IntervalMetadata)
+def _gff_to_metadata(fh, rec_num=1, **kwargs):
+    record = _get_nth_record(_parse_records(fh), rec_num)
+    return _construct(record, IntervalMetadata, **kwargs)
+
+
+def _parse_records(fh):
+    seqID = False
+    features = {}
+    for line in _line_generator(fh, skip_blanks=True, strip=True):
+        if not line.startswith('#'):
+            tabs = line.split('\t')
+
+            # extract annotations
+            _headers = tabs[:3] + tabs[5:8]
+
+            # extract attributes
+            _attributes = tabs[8]
+
+            # extract intervals
+            _intervals = tabs[3:5]
+
+            intervals = tuple(map(_parse_required, _intervals))
+
+            attr = _parse_attr(_attributes)
+            _annot = list(map(_parse_required, _headers))
+            _annot.append(attr)
+            annotation = Feature(dict(zip(_ANNOTATION_HEADERS, _annot)))
+
+            if _annot[0] == seqID or seqID is False:
+                seqID = _annot[0]
+                features[annotation] = intervals
+            else:
+                seqID = False
+                yield features
+                features = {}
+                features[annotation] = intervals
+
+    yield features
+
+
+def _attr_to_list(attr_list):
+    _tags = []
+    if any(isinstance(el, tuple) for el in attr_list):
+        for _attr in attr_list:
+            _tags.append('='.join([_attr[0], _attr[1]]))
+        return ';'.join(_tags)
+    else:
+        return '='.join([attr_list[0], attr_list[1]])
+
+
+@gff.writer(IntervalMetadata)
+def _IntervalMetadata_to_gff(obj, fh):
+    # write file header
+    fh.write('%s\n' % '##gff-version 3')
+
+    for features, interval in obj.features.items():
+        tab = [None] * 9
```

Comment:
better to replace `None` with `''`, because `str(None)` is evaluted to  `'None'` in your last line of code, which I suppose not what we want.